index.bs

```diff
@@ -1288,100 +1328,159 @@ partial interface ModelBuilder {
+dictionary MLWebGLTextureInput {
+  required WebGLTexture glTexture;
+  sequence<long> dimensions;
+};
 
-<script type=idl>
-dictionary Input {
-  required ArrayBufferView buffer;
+dictionary MLWebGLTextureOutput {
+  WebGLTexture glTexture;
+  sequence<long> dimensions;
+};
+
+dictionary MLWebGLBufferInput {
+  required WebGLBuffer glBuffer;
+  long offset;
+  long size;
+  sequence<long> dimensions;
+};
+
+dictionary MLWebGLBufferOutput {
+  WebGLBuffer glBuffer;
+  long offset;
+  long size;
+  sequence<long> dimensions;
+};
+
+dictionary MLGPUTextureInput {
+  required GPUTexture gpuTexture;
   sequence<long> dimensions;
 };
 
-dictionary Output {
-  ArrayBufferView buffer;
+dictionary MLGPUTextureOutput {
+  GPUTexture gpuTexture;
   sequence<long> dimensions;
 };
 
-typedef record<DOMString, Input> NamedInputs;
-typedef record<DOMString, Output> NamedOutputs;
+dictionary MLGPUBufferInput {
+  required GPUBuffer gpuBuffer;
+  long offset;
+  long size;
+  sequence<long> dimensions;
+};
+
+dictionary MLGPUBufferOutput {
+  GPUBuffer gpuBuffer;
+  long offset;
+  long size;
+  sequence<long> dimensions;
+};
+
+typedef record<DOMString, MLInput> MLNamedInputs;
+typedef record<DOMString, MLOutput> MLNamedOutputs;
+
+typedef record<DOMString, MLWebGLTextureInput> MLNamedWebGLTextureInputs;
+typedef record<DOMString, MLWebGLTextureOutput> MLNamedWebGLTextureOutputs;
+
+typedef record<DOMString, MLWebGLBufferInput> MLNamedWebGLBufferInputs;
+typedef record<DOMString, MLWebGLBufferOutput> MLNamedWebGLBufferOutputs;
+
+typedef record<DOMString, MLGPUTextureInput> MLNamedGPUTextureInputs;
+typedef record<DOMString, MLGPUTextureOutput> MLNamedGPUTextureOutputs;
+
+typedef record<DOMString, MLGPUBufferInput> MLNamedGPUBufferInputs;
+typedef record<DOMString, MLGPUBufferOutput> MLNamedGPUBufferOutputs;
 
 [SecureContext, Exposed=Window]
-interface Compilation {
-  Promise<NamedOutputs> compute(NamedInputs inputs, optional NamedOutputs outputs = {});
+interface MLCompilation {
+  Promise<MLNamedOutputs> compute(MLNamedInputs inputs, 
+                                  optional MLNamedOutputs outputs = {});
+
+  Promise<MLNamedWebGLOutputs> compute(MLNamedWebGLTextureInputs glTextureInputs, 
+                                       optional MLNamedWebGLTextureOutputs glTextureOutputs = {});
+
+  Promise<MLNamedWebGLOutputs> compute(MLNamedWebGLBufferInputs glBufferInputs, 
+                                       optional MLNamedWebGLBufferOutputs glBufferOutputs = {});
+
+  Promise<MLNamedGPUOutputs> compute(MLNamedGPUTextureInputs gpuTextureInputs,
+                                     optional MLNamedGPUTextureOutputs gpuTextureOutputs = {});
+
+  Promise<MLNamedGPUOutputs> compute(MLNamedGPUBufferInputs gpuBufferInputs,
+                                     optional MLNamedGPUBufferOutputs gpuBufferOutputs = {});
```

Comment:
With `MLBuffer`, we may also simplify the `compute` method by union types.

e.g.
```js
dictionary MLInput {
  required (MLBuffer or WebGLTexture or GPUTexture) data;
  sequence<long> dimensions;
};

dictionary MLOutput {
  (MLBuffer or WebGLTexture or GPUTexture) data;
  sequence<long> dimensions;
};

typedef record<DOMString, MLInput> MLNamedInputs;
typedef record<DOMString, MLOutput> MLNamedOutputs;

Promise<MLNamedOutputs> compute(MLNamedInputs inputs, 
                                optional MLNamedOutputs outputs = {});
```

Reply:
Likewise. I'll make that change.

Reply:
Looks good, thanks!